src/plugins/data/public/search/search_interceptor.ts

```diff
@@ -166,13 +172,17 @@ export class SearchInterceptor {
       timeoutController.abort();
     });
 
+    const externalAbortController = new AbortController();
+
     // Get a combined `AbortSignal` that will be aborted whenever the first of the following occurs:
     // 1. The user manually aborts (via `cancelPending`)
     // 2. The request times out
-    // 3. The passed-in signal aborts (e.g. when re-fetching, or whenever the app determines)
+    // 3. abort() is called on `externalAbortController`
+    // 4. The passed-in signal aborts (e.g. when re-fetching, or whenever the app determines)
     const signals = [
       this.abortController.signal,
       timeoutSignal,
+      externalAbortController.signal,
```

Comment:
This new signal is used to cancel specific search.
Session service is using it to cancel only searches that belong to current session.

Reply:
Could we add that comment into the 3rd code comment? "Abort() is called on `externalAbortController`" isn't really self-explanatory without more details.